DedicatedServer/ModEntry.cs

```diff
@@ -22,20 +23,59 @@ public class ModEntry : Mod
         
         // TODO Consider what the automated host should do when another player proposes to them.
 
+        private WaitCondition titleMenuWaitCondition;
+        private ModConfig config;
+
         /*********
         ** Public methods
         *********/
         /// <summary>The mod entry point, called after the mod is first loaded.</summary>
         /// <param name="helper">Provides simplified APIs for writing mods.</param>
         public override void Entry(IModHelper helper)
         {
-            var config = helper.ReadConfig<ModConfig>();
-            new StartFarmStage(helper, Monitor, config).Enable();
-            helper.Events.GameLoop.UpdateTicked += PrintDebug;
+            this.config = helper.ReadConfig<ModConfig>();
+
+            // ensure that the game environment is in a stable state before the mod starts executing
+            this.titleMenuWaitCondition = new WaitCondition(() => Game1.activeClickableMenu is TitleMenu, 5);
```

Comment:
Hi, thanks for the work. I didn't realize that this could cause problems :-)
I tested it but had to enter the full name `StardewValley.Menus.TitleMenu` instead of `TitleMenu`.
Is this the correct namespace? Can you change it?

Reply:
Sorry, I've changed it